### DISCUSSION TITLE
typescript noImplicitAny not working in next js

### DISCUSSION THREAD

**Discussion by romanusname** [vercel/next.js#67551] [Help]
### Summary

typescript noImplicitAny not working in next js.

```
"strict": true,
"noImplicitAny": true,
```
example code:

```
function f1(param){
    let test = param * 2
    console.log(test)
}
f1("Hello")
```

in the typescript playground I get an error: https://www.typescriptlang.org/play/?#code/GYVwdgxgLglg9mABMAjACgA4EMBOWC2AlAN4BQiFiANgKZSJQ0DO9AvItnvogFSIBM5ShARM4tAHRU4AczSMWhUgF9SpVGgBEACRpVpmwkA

in my next js project the code compiles successfully. WebStorm doesn't indicate an error:
![Screenshot_31](https://github.com/vercel/next.js/assets/122212600/954d4c39-e1a2-46ee-94e5-ccda0d16edc5)

console log - NaN

### Additional information

```js
"next": "14.2.3",
```

**Comment by BorhanSaflo** [comment]
It seems like this issue might be related to your linter. Next.js uses [ESLint](https://nextjs.org/docs/app/building-your-application/configuring/eslint) for linting. Try running `npx next lint` to check if the linter identifies the error.

**Reply by romanusname** [reply]
Thanks for the advice! I did some more testing.
I started using VSCode instead of WebStorm. Now these problem areas are highlighted.
![Screenshot_37](https://github.com/vercel/next.js/assets/122212600/328f69b9-5ed4-4884-8fd0-d04997f8177e)


Now so that such code does not compile. I installed this:
`npm install --save-dev @typescript-eslint/parser @typescript-eslint/eslint-plugin`

And my .eslintrc.json is now like this:
```
{
  "extends": [
    "next/core-web-vitals",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "./tsconfig.json",
    "ecmaVersion": 2020,
    "sourceType": "module"
  },
  "plugins": ["@typescript-eslint"],
  "rules":{
    "@typescript-eslint/no-explicit-any": ["error"]
  }

}
```

In this case, the command - "npx next lint" still displays success in the console:
`✔ No ESLint warnings or errors`
